pkg/clients/dynatrace/dynatrace_client.go

```diff
@@ -146,11 +152,51 @@ func (dtc *dynatraceClient) makeRequestForBinary(ctx context.Context, url string
 	return hex.EncodeToString(hash.Sum(nil)), err
 }
 
-func (dtc *dynatraceClient) handleErrorResponseFromAPI(response []byte, statusCode int) error {
+func (dtc *dynatraceClient) handleErrorResponseFromAPI(response []byte, statusCode int, headers http.Header) error {
 	se := serverErrorResponse{}
+
+	contentType := "unknown"
+	proxy := ""
+
+	if headers != nil {
+		for _, field := range []string{"X-Forwarded-For", "Forwarded", "Via"} {
+			if value := headers.Get(field); value != "" {
+				proxy = value
+
+				break
+			}
+		}
+
+		if value := headers.Get("Content-Type"); value != "" {
+			contentType = value
+		}
+	}
+
 	if err := json.Unmarshal(response, &se); err != nil {
-		return errors.WithMessagef(err, "response error, can't unmarshal json response %d", statusCode)
+		var sb strings.Builder
+
+		sb.WriteString(fmt.Sprintf("Server returned status code %d", statusCode))
+
+		if proxy != "" {
+			sb.WriteString(fmt.Sprintf(" (via proxy %s)", proxy))
+		}
+
+		responseLen := int(math.Min(float64(GetMaxResponseLen()), float64(len(response))))
```

Comment:
```suggestion
		responseLen := min(GetMaxResponseLen(), len(response))
```
go supports generics ;)

Reply:
Done in 36ad638a